pkg/adapter/adapter.go

```diff
@@ -101,15 +100,15 @@ func (l *listener) Start(ctx context.Context) error {
 
 func (l listener) handleEvent(ctx context.Context) http.HandlerFunc {
 	return func(response http.ResponseWriter, request *http.Request) {
-		c := make(chan struct{})
-		go func() {
-			c <- struct{}{}
-			if err := l.run.WatchConfigMapChanges(ctx); err != nil {
-				log.Fatal("error from WatchConfigMapChanges for controller : ", err)
-			}
-		}()
-		// Force WatchConfigMapChanges go routines to actually start
-		<-c
+		// we should fix this, this basically reads configmap on every request, is that supposed to be okay?
```

Comment:
i think switching to viper will fix this

Reply:
yes that's the idea :)